sphinx_js/doclets.py

```diff
@@ -84,6 +86,29 @@ def make(self):
         return [self.program] + self.args
 
 
+def cache_to_file(get_filename):
+    """Returns a decorator that will cache the result of fn to a file
+    
+    The name of the file is computed at run-time by get_filename.
+    get_filename receives the original arguments of fn.
+    """
+    def decorator(fn):
+        @functools.wraps(fn)
+        def decorated(*args, **kwargs):
+            filename = get_filename(*args, **kwargs)
+            if filename and os.path.isfile(filename):
+                with open(filename) as f:
+                    return load(f)
+            res = fn(*args, **kwargs)
+            if filename:
+                with open(filename, 'w+') as f:
```

Comment:
It shouldn't technically matter since you arrive here only if the file doesn't exist, but why not use the more common "w" mode rather than the appending "w+"?

Reply:
If I am not mistaken, the "+" is to automatically create the file if it does not exist, the "a" mode is for appending.

Reply:
Looks like we were both wrong! "+" opens the file for reading *and* writing. "a" is indeed for appending. "w" is for truncating. So I think "w" is the best fit here.

Reply:
Thanks, looks like we are not the only ones being confused : https://stackoverflow.com/questions/16208206/confused-by-python-file-mode-w/16208298, I am updating the PR.